frontend/src/component/admin/network/NetworkTrafficUsage/chart-functions.ts

```diff
@@ -0,0 +1,136 @@
+import type { ChartDataset } from 'chart.js';
+import type { TrafficUsageDataSegmentedCombinedSchema } from 'openapi';
+import { endpointsInfo } from './endpoint-info';
+import {
+    addDays,
+    addMonths,
+    differenceInCalendarDays,
+    differenceInCalendarMonths,
+} from 'date-fns';
+import { formatDay, formatMonth } from './dates';
+import type { ChartDataSelection } from './chart-data-selection';
+export type ChartDatasetType = ChartDataset<'bar'>;
+
+// todo: test
+export const toChartData = (
+    traffic?: TrafficUsageDataSegmentedCombinedSchema,
+): { datasets: ChartDatasetType[]; labels: (string | number)[] } => {
+    if (!traffic) {
+        return { labels: [], datasets: [] };
+    }
+
+    if (traffic.grouping === 'monthly') {
+        return toMonthlyChartData(traffic);
+    } else {
+        return toDailyChartData(traffic);
+    }
+};
+
+type SegmentedSchemaApiData =
+    TrafficUsageDataSegmentedCombinedSchema['apiData'][0];
+
+// todo: integrate filtering `filterData` frontend/src/component/admin/network/NetworkTrafficUsage/util.ts
+const prepareApiData = (
```

Comment:
Since you've implemented the filtering here, the overage calculation could just be fed this output instead of the output from toChartData